<launch>
    <!-- Launches some of the base utils needed to get the planner to run -->
    <arg name="sim" default="true" />
    <arg name="ip" default="127.0.0.1" />
    <arg name="robot" default="ur5" />
    <arg name="use_camera" default="false" />
    <arg name="debug" default="$(arg sim)" />

    <param name="use_camera" value="$(arg use_camera)" />
    <param name="robot_ip" value="$(arg ip)" />
<!--    <param name="tool_speed" value="0.160" />-->
    <param name="debug" value="$(arg debug)" />

    <param name="apple_diameter" value="0.08" />
    <param name="palm_frame" value="manipulator" />

    <group unless="$(arg sim)">
        <include file="$(find ur_modern_driver)/launch/ur5e_bringup.launch">
            <arg name="robot_ip" value="$(arg ip)" />
        </include>
        <include file="$(find localisation)/launch/localisation.launch">
            <arg name="robot_name" value="$(arg robot)" />
        </include>
    </group>
    <group if="$(arg sim)">
        <include file="$(find localisation)/launch/localisation_sim.launch">
            <arg name="robot_name" value="$(arg robot)" />
            <arg name="use_camera" value="$(arg use_camera)" />
        </include>
    </group>
    <group if="$(arg use_camera)">
        <node name="pc_corrector" pkg="sawyer_planner" type="point_cloud_corrector.py" />
        <node name="map_camera_publisher" pkg="tf" type="static_transform_publisher" args="0.0266 0.0475 -0.10 0 0 0 1 or_ee_link camera_link 10" />
        <include file="$(find realsense2_camera)/launch/rs_rgbd.launch" />
<!--        <include file="$(find rgb_segmentation)/launch/rgb_segmentation.launch" />-->
<!--        <include file="$(find sawyer_planner)/launch/pcl_filter_nodelet.launch" />-->
<!--        <node name="background_detector" pkg="sawyer_planner" type="background_detector.py" />-->
<!--        <node name="point_tracker" pkg="sawyer_planner" type="point_tracker.py" unless="$(arg sim)" output="screen" />-->
<!--        <node name="branch_orientation" pkg="sawyer_planner" type="branch_orientation.py" />-->
<!--        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find sawyer_planner)/include/cloud_display.rviz" />-->
        <remap from="image" to="/camera/color/image_raw" />
        <node name="image_display" pkg="image_view" type="image_view" />

    </group>

    <group unless="$(arg use_camera)">
        <node name="map_camera_publisher" pkg="tf" type="static_transform_publisher" args="0.0266 0.0625 -0.10 -0.5 0.5 -0.5 0.5 or_ee_link camera_color_optical_frame 10" />
    </group>

<!--    <node name="cutpoint_loc_publisher" pkg="tf" type="static_transform_publisher" args="0 -0.0205 -0.045 0 0 0 1 manipulator cutpoint 10" />-->
    <node name="freedrive_node" pkg="sawyer_planner" type="freedrive_node.py" />

<!--    <node name="arduino_communicator" pkg="sawyer_planner" type="arduino_communicator.py" />-->
</launch>